controller/Battery.php

```diff
@@ -112,18 +116,16 @@ public function editInstance()
         $myFormContainer = new \tao_actions_form_Instance($clazz, $battery);
 
         $myForm = $myFormContainer->getForm();
-        if($myForm->isSubmited()){
-            if($myForm->isValid()){
-
-                $values = $myForm->getValues();
-                // save properties
-                $binder = new \tao_models_classes_dataBinding_GenerisFormDataBinder($battery);
-                $battery = $binder->bind($values);
-                $message = __('Battery saved');
-
-                $this->setData('message', $message);
-                $this->setData('reload', true);
-            }
+        if ($myForm->isSubmited() && $myForm->isValid()) {
+            $values = $myForm->getValues();
+            // save properties
+            $binder = new \tao_models_classes_dataBinding_GenerisFormDataBinder($battery);
+            $battery = $binder->bind($values);
+            $message = __('Battery saved');
+            $this->getEventManager()->trigger(new BatteryModifiedEvent($battery, $values));
```

Comment:
Not sure, but consider using `$this->getClassService()->bindProperties()` (overload it in service and trigger event in it)

Reply:
Don't see a reason to do that. Can you elaborate?

Reply:
The same reason as for `createInstance` above

Reply:
I keep my state as above